test/s3select_test_json_parser.cpp

```diff
@@ -0,0 +1,586 @@
+	ASSERT_EQ( sax_exact_filter(TEST2, from_clause, projection) ,0);
+    
+	std::vector <std::string> from_clause1 = {"nested_obj"};
+	std::vector <std::string> projection1 = {"hello2"};
+	ASSERT_EQ( sax_exact_filter(TEST3, from_clause1, projection1) ,0);
+
+	std::vector <std::string> from_clause2 = {"nested_obj"};
+	std::vector <std::string> projection2{};
+	std::vector <std::string> where_clause2 = {"nested2", "c1"};
+	ASSERT_EQ( sax_exact_filter(TEST3, from_clause2, projection2, where_clause2) ,0);
+	
+	std::vector <std::string> from_clause3 = {"nested_obj"};
+	std::vector <std::string> projection3 = {"nested2", "array_nested2"};
+	ASSERT_EQ( sax_exact_filter(TEST3, from_clause3, projection3) ,0);
+
+	std::vector <std::string> from_clause4 = {"nested_obj"};
+	std::vector <std::string> projection4 = {"nested2", "c1"};
+	std::vector <std::string> where_clause4{"nested2", "array_nested2"};
+	ASSERT_EQ( sax_exact_filter(TEST3, from_clause4, projection4, where_clause4) ,0);
+	
+    std::vector <std::string> from_clause5 = {"nested_obj", "nested3"};
+	std::vector <std::string> projection5 = {"nested4", "c1"};
+	std::vector <std::string> where_clause5{"hello3"};
+	ASSERT_EQ( sax_exact_filter(TEST3, from_clause5, projection5, where_clause5) ,0);
+
+	std::vector <std::string> from_clause6 = {"nested_obj", "nested3"};
+	std::vector <std::string> projection6 = {"t2"};
+	std::vector <std::string> where_clause6 = {"nested4", "array_nested3"};
+	ASSERT_EQ( sax_exact_filter(TEST3, from_clause6, projection6, where_clause6) ,0);
+
+	std::vector <std::string> from_clause7 = {"glossary"};
+	std::vector <std::string> projection7 = {"title"};
+	ASSERT_EQ( sax_exact_filter(TEST4, from_clause7, projection7) ,0);
+
+	std::vector <std::string> from_clause8 = {"glossary"};
+	std::vector <std::string> projection8 = {"title"};
+	std::vector <std::string> where_clause8 = {"GlossDiv", "title"};
+	ASSERT_EQ( sax_exact_filter(TEST4, from_clause8, projection8, where_clause8) ,0);
+
+	std::vector <std::string> from_clause9 = {"glossary", "GlossDiv"};
+	std::vector <std::string> projection9 = {"GlossList", "GlossEntry", "GlossDef", "para"};
+	std::vector <std::string> where_clause9 = {"GlossList", "GlossEntry", "GlossDef", "GlossSeeAlso"};
+	ASSERT_EQ( sax_exact_filter(TEST4, from_clause9, projection9, where_clause9) ,0);
+
+	std::vector <std::string> from_clause10 = {"glossary", "GlossDiv"};
+	std::vector <std::string> projection10 = {"GlossList", "GlossEntry", "GlossDef", "postarray", "a"};
+	std::vector <std::string> where_clause10 = {"GlossList", "GlossEntry", "GlossSee"};
+	ASSERT_EQ( sax_exact_filter(TEST5, from_clause10, projection10, where_clause10) ,0);
+}
+	
+TEST(TestS3selectJsonParser, prefix_match)
+{
+	std::vector <std::string> from_clause = {"row", "color"};
+	ASSERT_EQ( sax_filter_results(TEST2, from_clause) ,0);
+
+	std::vector <std::string> from_clause1 = {"nested_obj"};
+	ASSERT_EQ( sax_filter_results(TEST3, from_clause1) ,0);
```

Comment:
`sax_filter_results` always returns 0
what is the purpose of the ASSERT?

Reply:
ASSERT don't have a purpose as of now. It prints the exact-match, prefix-match results